### PR TITLE
Use Devise gem

### PR DESCRIPTION
**Pull request by khanhnv3007**

### CODE REVIEW

File: app/assets/javascripts/static_pages.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (hnam2812) [inline app/assets/javascripts/static_pages.coffee]: remove file

File: app/assets/stylesheets/static_pages.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the static_pages controller here.
```
Comment (hnam2812) [inline app/assets/stylesheets/static_pages.scss]: remove file

File: app/helpers/static_pages_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
+end
```
Comment (hnam2812) [inline app/helpers/static_pages_helper.rb]: remove file

File: app/models/user.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 class User < ActiveRecord::Base
-  has_many :questions, dependent: :destroy
-  has_many :exams, dependent: :destroy
+  # Include default devise modules. Others available are:
```
Comment (hnam2812) [inline app/models/user.rb]: remove mark

File: app/views/static_pages/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<h1>StaticPages#index</h1>
```
Comment (hnam2812) [inline app/views/static_pages/index.html.erb]: remove file

File: Gemfile [hunk 6]
```diff
@@ -9,6 +9,7 @@ gem "coffee-rails", "~> 4.1.0"
 gem "jquery-rails"
 gem "turbolinks"
 gem "jbuilder", "~> 2.0"
+gem 'devise'
```
Comment (dieunb) [inline Gemfile]: nháy kép nhé

File: config/environments/development.rb [hunk 7]
```diff
@@ -36,6 +36,8 @@
   # Raises helpful error messages.
   config.assets.raise_runtime_errors = true
 
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
```
Comment (dieunb) [inline config/environments/development.rb]: convention

File: app/controllers/application_controller.rb [hunk 8]
```diff
@@ -1,5 +1,10 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+
```
Comment (hnam2812) [inline app/controllers/application_controller.rb]: remove this line

File: app/models/user.rb [hunk 9]
```diff
@@ -1,4 +1,7 @@
 class User < ActiveRecord::Base
   has_many :questions, dependent: :destroy
   has_many :exams, dependent: :destroy
+
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
```
Comment (hnam2812) [inline app/models/user.rb]: indent

**Comment by hnam2812** [comment]
OK

File: config/environments/development.rb [hunk 10]
```diff
@@ -36,6 +36,8 @@
   # Raises helpful error messages.
   config.assets.raise_runtime_errors = true
 
+  config.action_mailer.default_url_options = { host: "localhost", port: 3000 }
```
Comment (dieunb) [inline config/environments/development.rb]: Convention

**Pull request merged**